src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -119,47 +126,37 @@ def _assert_matching_indices(self, data_indices: dict) -> None:
             self._internal_output_idx,
         ), f"Model indices must match {self._internal_input_idx} != {self._internal_output_idx}"
 
-    def _define_tensor_sizes(self, config: DotConfig) -> None:
+    def _define_tensor_sizes(self, config: DotDict) -> None:
         # Define Sizes of different tensors
-        self._data_grid_size = self._graph_data[(self._graph_name_data, "to", self._graph_name_data)].ecoords_rad.shape[
-            0
-        ]
-        self._hidden_grid_size = self._graph_data[
-            (self._graph_name_hidden, "to", self._graph_name_hidden)
-        ].hcoords_rad.shape[0]
-
-        self.trainable_data_size = config.model.trainable_parameters.data
-        self.trainable_hidden_size = config.model.trainable_parameters.hidden
-
-    def _register_latlon(self, name: str, key: str) -> None:
+        self.num_nodes = {name: self._graph_data[name]["coords"].shape[0] for name in self._graph_mesh_names}
+        self.num_node_features = {
+            name: 2 * self._graph_data[name]["coords"].shape[1] for name in self._graph_mesh_names
+        }
+        self.num_trainable_params = {
+            name: config.model.trainable_parameters.get("data" if name != "hidden" else name, 0)
```

Comment:
I wonder if this should be hard-coded like this. Also should there be a warning if this sets 0?